src/main/java/Task.java

```diff
@@ -0,0 +1,136 @@
+        for (int i = 1; i < this.date.split("\\s").length; i++) {
+            if (i == 1) {
+                dateFormatted += this.date.split("\\s")[i];
+            } else {
+                dateFormatted += " " + this.date.split("\\s")[i];
+            }
+        }
+        LocalDate ld = LocalDate.parse(dateFormatted);
+        this.processedDate = ld;
+        this.hasProcessedDate = true;
+        this.fileDate = ld.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        return this.fileDate;
+    }
+
+    public String addProcessedDate(String dateFormatted) {
+        //System.out.println(dateFormatted);
+        int month = 0;
+        int day = Integer.parseInt(dateFormatted.split("\\s")[1]);
+        int year = Integer.parseInt(dateFormatted.split("\\s")[2]);
+        if (dateFormatted.split("\\s")[0].equals("Jan")) {
+            month = 1;
+        } else if (dateFormatted.split("\\s")[0].equals("Feb")) {
+            month = 2;
+        } else if (dateFormatted.split("\\s")[0].equals("Mar")) {
+            month = 3;
+        } else if (dateFormatted.split("\\s")[0].equals("Apr")) {
+            month = 4;
+        } else if (dateFormatted.split("\\s")[0].equals("May")) {
+            month = 5;
+        } else if (dateFormatted.split("\\s")[0].equals("Jun")) {
+            month = 6;
+        } else if (dateFormatted.split("\\s")[0].equals("Jul")) {
+            month = 7;
+        } else if (dateFormatted.split("\\s")[0].equals("Aug")) {
+            month = 8;
+        } else if (dateFormatted.split("\\s")[0].equals("Sep")) {
+            month = 9;
+        } else if (dateFormatted.split("\\s")[0].equals("Oct")) {
+            month = 10;
+        } else if (dateFormatted.split("\\s")[0].equals("Nov")) {
+            month = 11;
+        } else {
+            month = 12;
+        }
+        this.processedDate = LocalDate.of(year, month, day);
+        this.hasProcessedDate = true;
+        this.fileDate = this.processedDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        return this.fileDate;
+    }
+
+    public String getDescription() {
+        if (this.type.equals("todo")) {
+            return this.name;
+        } else if (this.type.equals("deadline")) {
+//            String dateFormatted = "";
+//            this.date.split("\\s");
+//            for (int i = 1; i < this.date.split("\\s").length; i++) {
+//                if (i == 1) {
+//                    dateFormatted += this.date.split("\\s")[i];
+//                }
+//                else {
+//                    dateFormatted += " " + this.date.split("\\s")[i];
+//                }
+//            }
+//            dateFormatted = "(by: " + dateFormatted + ")";
+//            return this.name + dateFormatted;
+            if (this.hasProcessedDate == false) {
+                return this.name + " (by: " + this.getProcessedDate() + ")";
+            } else {
+                return this.name + " (by: " + this.fileDate + ")";
```

Comment:
It might be better to give the user an error message instead of using the fileDate so that the user can input a valid date.